src/pages/execution/tabLoader.tsx

```diff
@@ -255,6 +261,73 @@ export default function tabLoader(tabBase: TabBase): TabData {
   };
 }
 
+const generateAlerts = (tasks) => {
```

Comment:
`AlertComponent` and `SummaryTabHead` call `generateAlerts` on the same data. Can we do some re-structuring to make use of one call.

Reply:
Good suggestion! We're going to pass states to the execution so that the summary dot can know its color and whether to show up or not.